magefile.go

```diff
@@ -115,18 +118,53 @@ func dockerConfigFileParams() mage.ConfigFileParams {
 	}
 }
 
+func keepPackages(types []string) map[mage.PackageType]struct{} {
+	keep := make(map[mage.PackageType]struct{})
+	for _, t := range types {
+		var pt mage.PackageType
+		if err := pt.UnmarshalText([]byte(t)); err != nil {
+			log.Printf("skipped filtering package type %s", t)
+			continue
+		}
+		keep[pt] = struct{}{}
+	}
+	return keep
+}
+
+func filterPackages(types string) {
+	var packages []mage.OSPackageArgs
+	keep := keepPackages(strings.Split(types, " "))
+	for _, p := range mage.Packages {
+		for _, t := range p.Types {
+			if _, ok := keep[t]; !ok {
+				continue
+			}
+			packages = append(packages, p)
+			break
```

Comment:
why the `break`? can't keep 2 types of the same package or am i reading wrong?

Reply:
right, there are many types per package but you want just one package once the type matches.  The consequence is that, in our case, if you pick `rpm` you also get `deb`.  I thought about filtering that down further but it didn't seem work it when really all we're after for now is 'build docker images' or 'build linux distro packages'